addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java

```diff
@@ -123,105 +126,118 @@ public void dispose() {
-                updateState(channelComfortTemp.getUID(), new DecimalType(device.getHkr().getKomfort()));
+                updateThingChannelState(thing, CHANNEL_MODE, new StringType(device.getHkr().getMode()));
+                updateThingChannelState(thing, CHANNEL_LOCKED,
+                        device.getHkr().getLock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED : OpenClosedType.OPEN);
+                updateThingChannelState(thing, CHANNEL_ACTUALTEMP,
+                        new DecimalType(HeatingModel.toCelsius(device.getHkr().getTist())));
+                final BigDecimal settemp = HeatingModel.toCelsius(device.getHkr().getTsoll());
+                if (HeatingModel.inCelsiusRange(settemp)) {
+                    thing.getConfiguration().put(THING_SETTEMP, settemp);
+                }
+                updateThingChannelState(thing, CHANNEL_SETTEMP, new DecimalType(settemp));
+                final BigDecimal ecotemp = HeatingModel.toCelsius(device.getHkr().getAbsenk());
+                thing.getConfiguration().put(THING_ECOTEMP, ecotemp);
+                updateThingChannelState(thing, CHANNEL_ECOTEMP, new DecimalType(ecotemp));
+                final BigDecimal comforttemp = HeatingModel.toCelsius(device.getHkr().getKomfort());
+                thing.getConfiguration().put(THING_COMFORTTEMP, comforttemp);
+                updateThingChannelState(thing, CHANNEL_COMFORTTEMP, new DecimalType(comforttemp));
+                updateThingChannelState(thing, CHANNEL_RADIATOR_MODE,
+                        new StringType(device.getHkr().getRadiatorMode()));
                 if (device.getHkr().getNextchange() != null) {
-                    Channel channelNextChange = thing.getChannel(CHANNEL_NEXTCHANGE);
                     if (device.getHkr().getNextchange().getEndperiod() == 0) {
-                        updateState(channelNextChange.getUID(), UnDefType.UNDEF);
+                        updateThingChannelState(thing, CHANNEL_NEXTCHANGE, UnDefType.UNDEF);
                     } else {
-                        Calendar calendar = Calendar.getInstance();
+                        final Calendar calendar = Calendar.getInstance();
                         calendar.setTime(new Date(device.getHkr().getNextchange().getEndperiod() * 1000L));
-                        updateState(channelNextChange.getUID(), new DateTimeType(calendar));
+                        updateThingChannelState(thing, CHANNEL_NEXTCHANGE, new DateTimeType(calendar));
                     }
-                    Channel channelNextTemp = thing.getChannel(CHANNEL_NEXTTEMP);
-                    updateState(channelNextTemp.getUID(),
-                            new DecimalType(device.getHkr().getNextchange().getTchange()));
+                    updateThingChannelState(thing, CHANNEL_NEXTTEMP,
+                            new DecimalType(HeatingModel.toCelsius(device.getHkr().getNextchange().getTchange())));
```

Comment:
Here we still use the `toCelsius` method what does it do, what is the input value?